app/src/main/java/org/oppia/android/app/settings/profile/SnackbarManager.kt

```diff
@@ -0,0 +1,74 @@
+package org.oppia.android.app.settings.profile
+
+import android.util.Log
+import android.view.View
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.android.synthetic.main.profile_edit_fragment.*
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import javax.inject.Inject
+
+class SnackbarManager @Inject constructor(private val snackbarController: SnackbarController) {
+  // Use SnackbarController for displaying the current snackbar, hiding it after the snackbar hides, or showing a new one.
+
+  fun showSnackbar(@StringRes messageStringId: Int, duration: SnackbarController.SnackbarDuration) {
+    // Enqueue snackbar using SnackbarController...
+    snackbarController.enqueueSnackbar(
+      SnackbarController.SnackbarRequest.ShowSnackbar(
+        messageStringId,
+        duration
+      )
+    )
+  }
+
+  fun enableShowingSnackbars(activity: AppCompatActivity) {
+    // Called in an activity's onCreate to start listening for snackbar state.
+    snackbarController.getCurrentSnackbar().toLiveData().observe(activity) { result ->
+      when (result) {
+        is AsyncResult.Success -> when (val request = result.value) {
+          is SnackbarController.SnackbarRequest.ShowSnackbar -> showSnackbar(activity.view, request)
```

Comment:
`showSnackbar(activity.view, request)` didn't seem to work for me so I used `showSnackbar(activity.findViewById(android.R.id.content), request)`.

Reply:
okay